File: 2020/day3/answer.py
def countTrees(map, right, down):
  col = 0
  trees = 0
  width = len(map[0])

  for row in range(down, len(map), down):
    col += right

    if map[row][col % width] == '#':
      trees += 1

  return trees

with open('input') as f:
  input = f.read().split('\n')
  
  print(f'Part 1: {countTrees(input, 3, 1)}')
  print(f'Part 2: {countTrees(input, 1, 1) * countTrees(input, 3, 1) * countTrees(input, 5, 1) * countTrees(input, 7, 1) * countTrees(input, 1, 2)}')

File: 2020/javascript/day3/answer.js
const fs = require('fs')

const input = fs
    .readFileSync('input', { encoding: 'utf8', flag: 'r' })
    .split('\n')

const countTrees = (map, right, down) =>
    map
        .filter((_, index) => index && index % down == 0)
        .reduce(
            (count, row, index) =>
                row[((index + 1) * right) % row.length] == '#'
                    ? count + 1
                    : count,
            0
        )

console.log(`Part 1: ${countTrees(input, 3, 1)}`)
console.log(
    `Part 2: ${
        countTrees(input, 1, 1) *
        countTrees(input, 3, 1) *
        countTrees(input, 5, 1) *
        countTrees(input, 7, 1) *
        countTrees(input, 1, 2)
    }`
)
